src/nickelclock.cc

```diff
@@ -243,19 +264,40 @@ static void add_time_to_footer(ReadingFooter *rf, TimePos position)
     }
 }
 
-// On recent 4.x firmware versions, the header and footer are setup in 
-// Ui_ReadingView::setupUi(). They are ReadingFooter widgets, with names set to 
-// "header" and "footer". This makes it easy to find them with findChild().
-extern "C" __attribute__((visibility("default"))) void _nc_set_header_clock(ReadingView *_this) 
+static ReadingFooter* find_reading_footer(ReadingView *_this)
 {
     auto containerName = (nc_settings->placement() == TimePlacement::Header)
                          ? "header" : "footer";
 
     // Find header or footer
-    ReadingFooter *rf = _this->findChild<ReadingFooter*>(containerName);
+    auto rf = _this->findChild<ReadingFooter*>(containerName);
     if (!rf)
         nh_log("ReadingFooter '%s' not found in ReadingView", containerName);
+    return rf;
+}
 
+// On recent 4.x firmware versions, the header and footer are setup in 
+// Ui_ReadingView::setupUi(). They are ReadingFooter widgets, with names set to 
+// "header" and "footer". This makes it easy to find them with findChild().
+extern "C" __attribute__((visibility("default"))) void _nc_set_header_clock(ReadingView *_this) 
+{
+    // Find header or footer
+    ReadingFooter *rf = find_reading_footer(_this);
     add_time_to_footer(rf, nc_settings->position());
     ReadingView__ReaderIsDoneLoading(_this);
 }
+
+// The original method polishes the widget and it's childrent, and seems to 
+// reset margins to default. Therefore, hook and change the margins after it 
+// has done it's work.
+extern "C" __attribute__((visibility("default"))) void _nc_dark_mode_changed(ReadingView *_this) 
+{
+    // Let the dark mode stuff happen first
+    ReadingView__darkModeChanged(_this);
+    // Find header or footer
```

Comment:
@pgaskin I'm making an assumption that calling the original method first is safe. Is this correct? Or will I have to find an alternative approach?

Reply:
That is OK, but I think there may be a better way to do this (something to do with polish, update, or paint).

Reply:
Thanks for the hints, I'll look into that.

Reply:
Oh, and assuming this works, to make the updates happen *after* a polish, either dispatch a lambda to the main thread, or use a custom signal/slot with a QueuedConnection.

Reply:
I don't think it will work :( The `ReadingView` never receives a polish event for it's child `ReadingFooters`, and when I look closer, Kobo seems to have created their own sort of polish in the `QObjectUtil` class.

Reply:
> Another option might be to try replacing the `ReadingFooter` stylesheet with a modified one. I'll try that tomorrow.

Yeah, that works. I'll close this and open a new PR with the new approach.